attribute_rule_calculation/CreatePointsAlongLine.js

```diff
@@ -0,0 +1,120 @@
+// This rule will create n number of point features along a line at equidistant intervals. Optionally add endpoints.
+
+// Number of midspan points to be created not including endpoints.
+var midspan_points = 2
+// Create endpoints. Default is true.
+var create_endpoints = true;
+// Point class name.
+var point_class = "points"
+
+// *************       Functions            *************
+
+function drop_points_on_line(line, points_number) {
+    var mispan = [];
+    var distance_along_line = 0;
+    var break_distance = Length(line) / (points_number + 1)
+
+    // Handle multipart geometries.
+    for (var part in Geometry(line).paths) {
+        var line_part = Geometry(line).paths[part]
+
+        // loop through segments of line
+        for (var i in line_part) {
+            if (i == 0) continue;
+
+            // Construct a 2-point line segment from the current and previous point
+            var first_point = line_part[i-1];
+            var second_point = line_part[i]
+            var two_point_line = Polyline({'paths': [[[first_point.x, first_point.y], [second_point.x, second_point.y]]], 'spatialReference' : first_point.spatialReference});
+            var seg_length = Length(two_point_line);
+
+            if (seg_length + distance_along_line < break_distance) {
+                // Don't need to drop point along this segment. Add distance and continue.
+                distance_along_line += seg_length;
+                continue
+            }
+            if (seg_length + distance_along_line == break_distance) {
+                // Break point is endpoint of segment
+                push(mispan, Point(second_point))
+                distance_along_line = 0
+                continue
+            }
+
+            // Drop new points along segment. Account for segment being longer than break distance.
+            for (var current_length = seg_length + distance_along_line; current_length>=break_distance; ) {
+                var xy = get_xy_along_line(first_point.x, first_point.y, second_point.x, second_point.y, break_distance - distance_along_line)
+                var new_point = Point({"x": xy[0], "y": xy[1], "spatialReference": first_point.spatialReference})
+                push(mispan, new_point)
+                if (Count(mispan) == points_number) {
+                    return mispan
+                }
+                // reset values for next loop if remaining portion of segment is larger than break distance.
+                current_length = Length(Polyline({"paths": [[new_point, [second_point.x, second_point.y]]], "spatialReference": second_point.spatialReference}))
+                distance_along_line = 0
+                first_point = new_point
+            }
+            // exit early if we hit the midspan number
+            if (Count(mispan) == points_number) {
```

Comment:
move to start of loop to handle when a point is at the end of a internal segment